src/JustSaying/AwsTools/MessageHandling/SqsMessagePublisher.cs

```diff
@@ -112,4 +114,121 @@ public InterrogationResult Interrogate()
             QueueUrl
         });
     }
+
+    public Task PublishAsync(IEnumerable<Message> messages, CancellationToken cancellationToken)
+        => PublishAsync(messages, null, cancellationToken);
+
+    public async Task PublishAsync(IEnumerable<Message> messages, PublishBatchMetadata metadata, CancellationToken cancellationToken)
+    {
+        if (QueueUrl is null)
+        {
+            throw new PublishException("Queue URL was null, perhaps you need to call `StartAsync` on the `IMessagePublisher` before publishing.");
+        }
+
+        var size = metadata?.BatchSize ?? 10;
+        size = Math.Min(size, 10);
+
+        foreach (var chuck in messages.Chunk(size))
+        {
+            var request = BuildSendMessageBatchRequest(chuck, metadata);
+            SendMessageBatchResponse response;
+            try
+            {
+                response = await _client.SendMessageBatchAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+            catch (AmazonServiceException ex)
+            {
+                throw new PublishBatchException(
+                    $"Failed to publish batch message to SQS. {nameof(request.QueueUrl)}: {request.QueueUrl}",
+                    ex);
+            }
+
+            if (response != null)
+            {
+                using (_logger.BeginScope(new Dictionary<string, string>
+                {
+                    ["AwsRequestId"] = response.ResponseMetadata?.RequestId
+                }))
+                {
+                    if (response.Successful.Count > 0)
```

Comment:
To remove the need to redundant enumeration of the response if logging isn't enabled at this level for this logger:

```suggestion
                    if (response.Successful.Count > 0 && _logger.IsEnabled(LogLevel.Information))
```